src/frontend/src/telemetry.rs

```diff
@@ -12,16 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::sync::OnceLock;
+
 use prost::Message;
 use risingwave_common::telemetry::pb_compatible::TelemetryToProtobuf;
-use risingwave_common::telemetry::report::TelemetryReportCreator;
+use risingwave_common::telemetry::report::{report_event_common, TelemetryReportCreator};
 use risingwave_common::telemetry::{
     current_timestamp, SystemData, TelemetryNodeType, TelemetryReportBase, TelemetryResult,
 };
+use risingwave_pb::telemetry::{PbTelemetryDatabaseComponents, PbTelemetryEventStage};
 use serde::{Deserialize, Serialize};
 
 const TELEMETRY_FRONTEND_REPORT_TYPE: &str = "frontend";
 
+static FRONTEND_TELEMETRY_SESSION_ID: OnceLock<String> = OnceLock::new();
+static FRONTEND_TELEMETRY_TRACKING_ID: OnceLock<String> = OnceLock::new();
+
+pub fn set_frontend_telemetry_tracking_id_and_session_id(tracking_id: String, session_id: String) {
+    FRONTEND_TELEMETRY_TRACKING_ID.set(tracking_id).unwrap();
+    FRONTEND_TELEMETRY_SESSION_ID.set(session_id).unwrap();
+}
+
+fn _get_frontend_telemetry_tracking_id_and_session_id() -> (Option<String>, Option<String>) {
```

Comment:
yes, I am not sure we want such attr. `#[allow(dead_code)]` cause some trouble in source, and @xxchan complained about it.

Reply:
I complained using `#![allow(dead_code)]` **at the top-level**. 
- Using it on individual functions is ok. 
- Whether top-level or function-level, `#[expect(dead_code, reason="xxx")]` is better. Because it tells the reason, and tells you to remove it when the lint is no longer fullfilled.